scripts/ci/windows_flake_rate_proof.py

```
@@ -0,0 +1,305 @@
+    def __init__(self, command: list[str] | None, max_starts: int = 20) -> None:
+        self.command = command
+        self.max_starts = max_starts
+        self.starts = 0
+        self.process: subprocess.Popen | None = None
+
+    def keep_alive(self) -> None:
+        if self.command is None or self.starts >= self.max_starts:
+            return
+        if self.process is None or self.process.poll() is not None:
+            self.starts += 1
+            self.process = subprocess.Popen(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)